src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -113,6 +113,21 @@ protected void AddVariablesToEnvironment()
                     formattedKey,
                     pair.Value);
             }
+
+            if (!excludeSecrets)
+            {
+                // Add the secret variables to the environment variable dictionary prefix with Secret_.
+                foreach (KeyValuePair<string, string> pair in ExecutionContext.Variables.Private)
+                {
+                    // Format all variables other than "agent.jobstatus".
+                    string formattedKey = string.Equals(pair.Key, Constants.Variables.Agent.JobStatus, StringComparison.OrdinalIgnoreCase)
```

Comment:
You may want to yank out the part related to "agent.jobstatus" for the following reasons:
1) If that is set as a secret, then the customer did something wrong.
2) We're already prepending SECRET_ so it won't be found in it's normal location anyway.

I'm fine leaving it in as well if that's your preference.